hub/core/dataset.py

```diff
@@ -203,9 +222,22 @@ def create_tensor(
 
         if tensor_exists(name, self.storage):
             raise TensorAlreadyExistsError(name)
-        if name in vars(self):
+
+        if name in dir(self):
             raise InvalidTensorNameError(name)
 
+        if self.is_group():
+            return self.root.create_tensor(
+                posixpath.join(self.group_index, name),
+                htype,
+                dtype,
+                sample_compression,
+                chunk_compression,
+            )
+
+        if "/" in name:
```

Comment:
can we do a replace for "\\"s to "/"s? this was a problem for some windows users using directory structures

Reply:
We use posixpath instead of os.path, so same behavior on all platforms. P.S I develop on windows.